npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -64,6 +65,22 @@ def instrument_package_manager_version
         )
       end
 
+      def yarn_version
+        @yarn_version ||= begin
+          package = JSON.parse(package_json.content)
+          if (pkgmanager = package.fetch("packageManager", nil))
+            @yarn_version = get_yarn_version_from_path(pkgmanager)
+          elsif yarn_lock
+            @yarn_version = 1
+          end
+        end
+      end
+
+      def get_yarn_version_from_path(path)
+        version_match = path.match(/yarn@(?<version>\d+.\d+.\d+)/)
+        version_match.named_captures.fetch("version")
```

Comment:
❓ What kinds of guarantees do we have about `path` here? I can foresee some `NoMethodError`s being thrown if we don't have a full match, and a `KeyError` if the named captures aren't properly saturated.